frame/staking/src/mock.rs

```diff
@@ -16,47 +16,65 @@
-use sp_runtime::traits::{IdentityLookup, Convert, OpaqueKeys, OnInitialize, SaturatedConversion};
-use sp_runtime::testing::{Header, UintAuthorityId};
-use sp_staking::{SessionIndex, offence::{OffenceDetails, OnOffenceHandler}};
-use sp_core::{H256, crypto::key_types};
-use sp_io;
+use crate::*;
 use frame_support::{
-	assert_ok, impl_outer_origin, parameter_types, StorageLinkedMap, StorageValue,
-	traits::{Currency, Get, FindAuthor},
+	assert_ok, impl_outer_dispatch, impl_outer_event, impl_outer_origin, parameter_types,
+	traits::{Currency, FindAuthor, Get, PredictNextSessionChange},
 	weights::Weight,
+	StorageLinkedMap, StorageValue,
+};
+use frame_system::offchain::{CreateTransaction, Signer, TransactionSubmitter};
+use sp_core::{crypto::key_types, H256};
+use sp_io;
+use sp_phragmen::{build_support_map, evaluate_support, reduce, ExtendedBalance, StakedAssignment};
+use sp_runtime::curve::PiecewiseLinear;
+use sp_runtime::testing::{Header, TestXt, UintAuthorityId};
+use sp_runtime::traits::{
+	Convert, Extrinsic as ExtrinsicT, IdentityLookup, OnInitialize, OpaqueKeys,
+	SaturatedConversion, Zero,
 };
-use crate::{
-	EraIndex, GenesisConfig, Module, Trait, StakerStatus, ValidatorPrefs, RewardDestination,
-	Nominators, inflation
+use sp_runtime::{KeyTypeId, Perbill};
+use sp_staking::{
+	offence::{OffenceDetails, OnOffenceHandler},
+	SessionIndex,
 };
+use std::{cell::RefCell, collections::HashSet};
 
 /// The AccountId alias in this test module.
-pub type AccountId = u64;
-pub type BlockNumber = u64;
-pub type Balance = u64;
+pub(crate) type AccountId = u64;
+pub(crate) type AccountIndex = u64;
+pub(crate) type BlockNumber = u64;
+pub(crate) type Balance = u64;
+
+pub const PHRASE: &str = "news slush supreme milk chapter athlete soap sausage put clutch what kitten";
 
 /// Simple structure that exposes how u64 currency can be represented as... u64.
 pub struct CurrencyToVoteHandler;
 impl Convert<u64, u64> for CurrencyToVoteHandler {
-	fn convert(x: u64) -> u64 { x }
+	fn convert(x: u64) -> u64 {
+		x
+	}
 }
 impl Convert<u128, u64> for CurrencyToVoteHandler {
-	fn convert(x: u128) -> u64 { x.saturated_into() }
+	fn convert(x: u128) -> u64 {
+		x.saturated_into()
+	}
 }
 
 thread_local! {
 	static SESSION: RefCell<(Vec<AccountId>, HashSet<AccountId>)> = RefCell::new(Default::default());
+	static SESSION_PER_ERA: RefCell<SessionIndex> = RefCell::new(3);
 	static EXISTENTIAL_DEPOSIT: RefCell<u64> = RefCell::new(0);
 	static SLASH_DEFER_DURATION: RefCell<EraIndex> = RefCell::new(0);
+	static ELECTION_LOOKAHEAD: RefCell<BlockNumber> = RefCell::new(0);
+	static PERIOD: RefCell<BlockNumber> = RefCell::new(1);
+	static LOCAL_KEY_ACCOUNT: RefCell<AccountId> = RefCell::new(10);
 }
 
 pub struct TestSessionHandler;
 impl pallet_session::SessionHandler<AccountId> for TestSessionHandler {
-	const KEY_TYPE_IDS: &'static [KeyTypeId] = &[key_types::DUMMY];
+	// EVEN if no tests break, I must have broken something here... TODO
```

Comment:
Sadly all the tests do work and we need to figure out if anything is broken or not. 

In general, since part of the tests are using UIntAuthority and part of them a real sr25519 key, it might be worthwhile to even duplicate the mock file into two versions.